### ISSUE TITLE
Unexpected alias added by where.missing(...)

### ISSUE THREAD

**Issue by zhuravel** [rails/rails#47909]
Hi. I've upgrading my project to use unreleased Rails from `main`, and got an error.

### Steps to reproduce
```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  if ARGV[0] == "main"
    gem "rails", github: "rails/rails", branch: "main"
  else
    gem "activerecord", "7.0.4.3"
  end

  gem "sqlite3"
end

require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table(:books, force: true) do |t|
    t.references :progress
  end

  create_table(:progresses, force: true) do |t|
    t.references :book
    t.integer :percentage, default: 0, null: false
  end
end

class Book < ActiveRecord::Base
  has_one :progress

  scope :without_progress, -> do
    where.missing(:progress)
  end

  scope :completed, -> do
    joins(:progress).merge(Progress.completed)
  end
end

class Progress < ActiveRecord::Base
  scope :completed, -> do
    where(percentage: 100)
  end
end

class BugTest < Minitest::Test
  def test_bug
    assert_equal 0, Book.completed.count
    assert_equal 0, Book.without_progress.count
    assert_equal 0, Book.without_progress.completed.count
  end
end
```

The last assertion passes on 7.0.4.3 and fails on `main` because `where.missing(:progress)` adds a table alias in `main`.
See query produced by `Book.where.missing(:progress)`:
```sql
-- Rails main
SELECT "books".* FROM "books" LEFT OUTER JOIN "progresses" "progress" ON "progress"."book_id" = "books"."id" WHERE "progress"."id" IS NULL

-- Rails 7.0.4.3
SELECT "books".* FROM "books" LEFT OUTER JOIN "progresses" ON "progresses"."book_id" = "books"."id" WHERE "progresses"."id" IS NULL
```

### Expected behavior
Rails 7.0.4.3 does not add alias for "progresses" table and returns 0:
```sql
SELECT COUNT(*) FROM "books" INNER JOIN "progresses" ON "progresses"."book_id" = "books"."id" WHERE "progresses"."id" IS NULL AND "progresses"."percentage" = ?
```

### Actual behavior
Rails `main` aliases "progresses" to "progress" and raises exception:
```sql
SELECT COUNT(*) FROM "books" INNER JOIN "progresses" "progress" ON "progress"."book_id" = "books"."id" WHERE "progress"."id" IS NULL AND "progresses"."percentage" = ?
```
Error: `ActiveRecord::StatementInvalid: SQLite3::SQLException: no such column: progresses.percentage`

### System configuration
**Rails version**: main (7.1.0.alpha)

**Ruby version**: `ruby 3.2.2 (2023-03-30 revision e51014f9c0) [arm64-darwin22]`

**Comment by zhuravel** [comment]
@fatkodima After some debugging, found that this is due to change from #45015

**Comment by fatkodima** [comment]
Thanks for the report! I will look into that soon.

**Comment by mikdiet** [comment]
I also face this alias for has_on issue in the head of 7-0-stable

**Comment by tmimura39** [comment]
I have submitted a new issue because the recently released v7.0.5 seemed to contain this issue.
https://github.com/rails/rails/issues/48334

Because of this, I am unable to update my application to v7.0.5.
Please backport or release a new version.

Thank you.
